backends/arm/test/ops/test_mean_dim.py

```diff
@@ -16,42 +15,66 @@
 from executorch.exir.backend.backend_details import CompileSpec
 from parameterized import parameterized
 
-logger = logging.getLogger(__name__)
-logger.setLevel(logging.INFO)
-
-test_data_suite = [
-    # (test_name, test_data)
-    (
-        "zeros",
-        torch.zeros(1, 1280, 7, 7),
-    ),
-    (
-        "ones",
-        torch.ones(1, 1280, 7, 7),
-    ),
-    (
-        "rand",
-        torch.rand(1, 1280, 7, 7),
-    ),
-    (
-        "randn",
-        torch.randn(1, 1280, 7, 7),
-    ),
-]
-
 
 class TestMeanDim(unittest.TestCase):
     """Tests MeanDim, called AdaptiveAvgPool2d in Pytorch."""
 
-    class MeanDim(torch.nn.Module):
+    class AdaptiveAveragePool2d(torch.nn.Module):
+        test_data_suite = [
+            # (test_name, test_data)
+            (
+                "zeros",
+                torch.zeros(1, 1280, 7, 7),
+            ),
+            (
+                "ones",
+                torch.ones(1, 1280, 7, 7),
+            ),
+            (
+                "rand",
+                torch.rand(1, 1280, 7, 7),
+            ),
+            (
+                "randn",
+                torch.randn(1, 1280, 7, 7),
+            ),
+        ]
+
         def __init__(self):
             super().__init__()
-            self.mean_dim = torch.nn.AdaptiveAvgPool2d(output_size=(1, 1))
+            self.adaptive_avg_pool2d = torch.nn.AdaptiveAvgPool2d(output_size=(1, 1))
 
         def forward(self, x):
-            return self.mean_dim(x)
+            return self.adaptive_avg_pool2d(x)
 
-    def _test_meandim_tosa_MI_pipeline(
+    class MeanDim(torch.nn.Module):
+        test_data_suite = [
+            # (test_name, test_data)
+            ("zeros", torch.zeros(1, 1280, 7, 7), -1, True),
+            ("ones", torch.ones(1, 1280, 7, 7), (-1, 2), True),
+            (
+                "rand",
+                torch.rand(1, 1280, 7, 7),
+                (-1),
+                True,
+            ),
+            (
+                "randn",
+                torch.randn(1, 1280, 7, 7),
+                (-1, -2, -3),
+                True,
+            ),
+        ]
+
+        def __init__(self, dim: int | list[int] = -1, keepdim: bool = True):
+            super().__init__()
+            self.dim = dim
+            self.keepdim = keepdim
+
+        def forward(self, x: torch.Tensor):
+            return x.mean(dim=self.dim, keepdim=self.keepdim)
```

Comment:
why add `keepdim` as a arg if we don't support keepdim=False? curious

Reply:
We may need to insert transposes if we have 4D data with `keepdim=False` and `permute_memory_to_nhwc=True.` One idea is to extend https://github.com/pytorch/executorch/blob/main/backends/arm/_passes/insert_squeeze_after_sum_pass.py to meandim and var as well. 

Also, the main goal of this PR is to support layernorm which works with `keepdim=True`.